app/logic/getSLInstructorTableData.py

```diff
@@ -0,0 +1,20 @@
+from app.models.course import Course
+from app.models.user import User
+from app.models.courseInstructor import CourseInstructor
+
+def getProposalData(user):
+    """Returns dictionary with data used to populate SL proposal table"""
+    courses = (Course.select()
+                     .where(CourseInstructor.user==user)
+                     .join(CourseInstructor))
+    courseDict = {} #any reason why this is a dictionary of dictionaries, wouldn't a list of dictionaries be easier to work with?
+    #I just thought it would be easier to read. But a list is a-ok too.
+    for course in courses:
+        otherInstructors = (CourseInstructor.select().where(CourseInstructor.course==course))
+        faculty = [f"{instructor.user.firstName} {instructor.user.lastName}" for instructor in otherInstructors]
+        courseDict[course.courseName] = {
+        "name":course.courseName,
+        "faculty": ', '.join(faculty),

```

Comment:
I wouldn't comma-separate the list here. Leave the data as intact as possible until you need to use it.

Reply:
Done! The whole list is passed in and jinja will do the join. When adding the data to each row.